packages/lesswrong/components/ultraFeed/UltraFeedUserDialog.tsx

```diff
@@ -0,0 +1,229 @@
+  dialogContent: {
+    padding: 0,
+    height: '100%',
+    display: 'flex',
+    flexDirection: 'column',
+    overflow: 'hidden',
+    '&:first-child': {
+      paddingTop: 0,
+    },
+  },
+  dialogPaper: {
+    width: '600px',
+    maxWidth: '90vw',
+    height: '80vh',
+    maxHeight: '80vh',
+    margin: '50px auto',
+    borderRadius: 8,
+    [theme.breakpoints.down('sm')]: {
+      width: '100vw',
+      maxWidth: '100vw',
+      height: '100dvh',
+      maxHeight: '100dvh',
+      margin: 0,
+      borderRadius: 0,
+    },
+  },
+  header: {
+    height: 56,
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    backgroundColor: theme.palette.background.pageActiveAreaBackground,
+    borderBottom: theme.palette.border.faint,
+    zIndex: theme.zIndexes.header,
+    padding: '8px 12px',
+    flexShrink: 0,
+    borderRadius: '8px 8px 0 0',
+    [theme.breakpoints.down('sm')]: {
+      borderRadius: 0,
+    },
+  },
+  headerLeft: {
+    display: 'flex',
+    alignItems: 'center',
+    gap: 12,
+    flex: 1,
+  },
+  headerTitle: {
+    ...theme.typography.body1,
+    fontWeight: 600,
+    color: theme.palette.text.primary,
+  },
+  '@keyframes fadeInHeader': {
+    from: {
+      opacity: 0,
+    },
+    to: {
+      opacity: 0.8,
+    },
+  },
+  headerUserInfo: {
+    display: 'flex',
+    alignItems: 'baseline',
+    justifyContent: 'space-between',
+    flex: 1,
+    opacity: 0.9,
+    animation: 'fadeInHeader 0.3s ease-in-out',
+  },
+  headerUsername: {
+    position: 'relative',
+    ...theme.typography.postStyle,
+    fontSize: '1.4em',
+    fontWeight: 600,
+    color: theme.palette.text.primary,
+    textDecoration: 'none',
+    '&:hover': {
+      opacity: 0.8,
+    },
+  },
+  headerFollowButton: {
+    marginLeft: 'auto',
+  },
+  headerButtonsContainer: {
+    display: 'flex',
+    gap: 8,
+    alignItems: 'center',
+    marginLeft: 'auto',
+  },
+  closeButton: {
+    width: 36,
+    height: 36,
+    color: theme.palette.grey[900],
+    backgroundColor: theme.palette.grey[200],
+    borderRadius: 4,
+    padding: 6,
+    cursor: 'pointer',
+    fontSize: 36,
+    fontWeight: 'bold',
+    '&:hover': {
+      color: theme.palette.grey[900],
+      backgroundColor: theme.palette.grey[300],
+    },
+    '& svg': {
+      display: 'block',
+      strokeWidth: 4.5,
+    },
+  },
+  contentWrapper: {
+    flex: 1,
+    display: 'flex',
+    flexDirection: 'column',
+    overflow: 'hidden',
+    width: '100%',
+  },
+  cardWrapper: {
+    width: '100%',
+    height: '100%',
+    display: 'flex',
+    flexDirection: 'column',
+    boxShadow: 'none',
+    borderRadius: 0,
+  },
+  modalWrapper: {
+    zIndex: `${theme.zIndexes.ultrafeedModal} !important`,
+  },
+}));
+
+const UltraFeedUserDialog = ({ 
+  user, 
+  onClose 
+}: {
+  user: UsersMinimumInfo;
+  onClose: () => void;
+}) => {
+  const classes = useStyles(styles);
+  const currentUser = useCurrentUser();
```

Comment:
This should be `useCurrentUserId`, which Jim recently implemented and causes it components that use it to only rerender if the currentUser._id changes, rather than any field on the currentUser.